megamek/src/megamek/common/Entity.java

```diff
@@ -141,6 +141,74 @@
         }
     }
 
+
+
+    public enum MPBoosters {
+        NONE(),
+        MASC_ONLY(),
+        SUPERCHARGER_ONLY(),
+        MASC_AND_SUPERCHARGER();
+
+        // common methods so we can change implementation if needed
+        public boolean hasMASCOnly()
+        {
+            return (this == MASC_ONLY);
+        }
+
+        public boolean hasSuperchargerOnly()
+        {
+            return (this == SUPERCHARGER_ONLY);
+        }
+
+
+        public boolean hasMASCXorSupercharger()
+        {
+            return (this == MASC_ONLY || this == SUPERCHARGER_ONLY);
+        }
+
+        // common methods so we can change implementation if needed
+        public boolean hasMASCAndOrSupercharger()
+        {
+            return (this != NONE);
+        }
+
+        public boolean hasMASCAndSupercharger()
+        {
+            return (this == MASC_AND_SUPERCHARGER);
+        }
+
+        public boolean hasMASC()
+        {
+            return (this == MASC_ONLY || this == MASC_AND_SUPERCHARGER);
+        }
+
+        public boolean hasSupercharger()
+        {
+            return (this == SUPERCHARGER_ONLY || this == MASC_AND_SUPERCHARGER);
+        }
+
+        public int calcRunMP(int walkMP) {
+            if (hasMASCXorSupercharger()) {
+                return (int) Math.ceil(walkMP * 2);
+            }
+            if (hasMASCAndSupercharger()) {
+                return (int) Math.ceil(walkMP * 2.5);
+            }
+            return (int) Math.ceil(walkMP * 1.5);
+        }
+
+        public int calcSprintMP(int walkMP) {
+            if (hasMASCXorSupercharger()) {
+                return (int) Math.ceil(walkMP * 2.5);
+            }
+            if (hasMASCAndSupercharger()) {
+                return (int) Math.ceil(walkMP * 3);
+            }
+            return 2;
```

Comment:
Can you elaborate on this 2 here? Seems like it should be walkMP * 2 instead.

Reply:
you are correct.  I will fix.. Currently Nothing that calls this will have neither which is why I missed it in testing.